File: app/src/main/kotlin/com/openclaw/ghostcrab/domain/model/config/GatewayMdnsSection.kt
package com.openclaw.ghostcrab.domain.model.config

import kotlinx.serialization.Serializable

/**
 * Typed representation of the `gateway.mdns` config sub-section.
 *
 * @param enabled Whether the gateway advertises itself via mDNS.
 * @param serviceName mDNS service name used for discovery.
 */
@Serializable
data class GatewayMdnsSection(
    val enabled: Boolean = true,
    val serviceName: String = "openclaw-gateway",
)

File: ios/GhostCrab/Domain/Model/Config/GatewayMdnsSection.swift
import Foundation

/// Typed representation of the `gateway.mdns` config sub-section.
///
/// - Parameters:
///   - enabled: Whether the gateway advertises itself via mDNS.
///   - serviceName: mDNS service name used for discovery.
public struct GatewayMdnsSection: Codable, Sendable, Hashable {
    public let enabled: Bool
    public let serviceName: String

    public init(enabled: Bool = true, serviceName: String = "openclaw-gateway") {
        self.enabled = enabled
        self.serviceName = serviceName
    }
}
